client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/RollupDocumentationIT.java

```diff
@@ -160,4 +173,164 @@ public void onFailure(Exception e) {
             assertTrue(latch.await(30L, TimeUnit.SECONDS));
         }
     }
+
+    public void testGetRollupCaps() throws Exception {
+        RestHighLevelClient client = highLevelClient();
+
+
+        DateHistogramGroupConfig dateHistogram =
+            new DateHistogramGroupConfig("timestamp", DateHistogramInterval.HOUR, new DateHistogramInterval("7d"), "UTC"); // <1>
+        TermsGroupConfig terms = new TermsGroupConfig("hostname", "datacenter");
+        HistogramGroupConfig histogram = new HistogramGroupConfig(5L, "load", "net_in", "net_out");
+        GroupConfig groups = new GroupConfig(dateHistogram, histogram, terms);
+        List<MetricConfig> metrics = new ArrayList<>(); // <1>
+        metrics.add(new MetricConfig("temperature", Arrays.asList("min", "max", "sum")));
+        metrics.add(new MetricConfig("voltage", Arrays.asList("avg", "value_count")));
+
+        //tag::x-pack-rollup-get-rollup-caps-setup
+        final String indexPattern = "docs";
+        final String rollupIndexName = "rollup";
+        final String cron = "*/1 * * * * ?";
+        final int pageSize = 100;
+        final TimeValue timeout = null;
+
+        String id = "job_1";
+        RollupJobConfig config = new RollupJobConfig(id, indexPattern, rollupIndexName, cron,
+            pageSize, groups, metrics, timeout);
+
+        PutRollupJobRequest request = new PutRollupJobRequest(config);
+        PutRollupJobResponse response = client.rollup().putRollupJob(request, RequestOptions.DEFAULT);
+
+        boolean acknowledged = response.isAcknowledged();
+        //end::x-pack-rollup-get-rollup-caps-setup
+        assertTrue(acknowledged);
+
+        // Now that the job is created, we should have a rollup index with metadata.
+        // We can test out the caps API now.
+
+        //tag::x-pack-rollup-get-rollup-caps-request
+        GetRollupCapsRequest getRollupCapsRequest = new GetRollupCapsRequest("docs");
+        //end::x-pack-rollup-get-rollup-caps-request
+
+        //tag::x-pack-rollup-get-rollup-caps-execute
+        GetRollupCapsResponse capsResponse = client.rollup().getRollupCapabilities(getRollupCapsRequest, RequestOptions.DEFAULT);
+        //end::x-pack-rollup-get-rollup-caps-execute
+
+        //tag::x-pack-rollup-get-rollup-caps-response
+        Map<String, RollableIndexCaps> rolledPatterns = capsResponse.getJobs();
+
+        RollableIndexCaps docsPattern = rolledPatterns.get("docs");
+
+        // indexName will be "docs" in this case... the index pattern that we rolled up
+        String indexName = docsPattern.getIndexName();
```

Comment:
Usually we use bullet points like `// <1>` to describe lines of code, would it make sense here too?

Reply:
So I went with inline comments because it's a longish snippet, and all the bullets would be located at the bottom under the snippet.  It felt better to describe as the user reads the code, rather than jumping back/forth to the bottom of the snippet each time?

I could try breaking the snippet up into shorter blocks, but that may make it more confusing since it's dealing with the same response object.